src/reactnativeapp/README.md

```diff
@@ -0,0 +1,86 @@
+# Example React Native app
+
+This was created using [`npx create-expo-app@latest`](https://reactnative.dev/docs/environment-setup#start-a-new-react-native-project-with-expo)
+
+Content was taken from the web app example in src/frontend and modified to work
+in a React Native environment.
+
+## Get started
+
+Start the OpenTelemetry demo:
+
+```bash
+cd ../..
+make start # or start-minimal
+```
+
+## Building the app
+
+Unlike the other components under src/ which run within docker containers this
+app must be built and then run on a mobile simulator on your machine or a physical
+device. If this is your first time running a React Native app then in order to
+execute the steps under "Build on your host machine" you will need to setup your
+local environment for Android or iOS development or both following
+[this guide](https://reactnative.dev/docs/set-up-your-environment). Alternatively
+for Android you can instead follow the steps under "Build within a container" to
+leverage a container to build the app's apk for you.
+
+### Build on your host machine
+
+Build and run the React Native app for a given platform by running these make targets
+from the project root:
+
+```bash
+make reactnative-run-android
+```
+
+Or
+
+```bash
+make reactnative-run-ios
+```
+
+You can also install dependencies and launch the app directly from this folder using:
+
+```bash
+cd src/reactnativeapp
+npm install
+```
+
+```bash
+npm run android
+```
+
+Or
+
+```bash
+npm run ios
+```
+
+Note that for all the above commands a release build is created rather than a debug
+one to avoid needing a dev server to provide the JS bundle.
+
+### Build within a container
+
+For Android builds you can produce an apk using Docker without requiring the dev
+tools to be installed on your host. From the project root run:
+
+```bash
+make reactnative-build-android
```

Comment:
Also got an error here:

```
android.Dockerfile:9
--------------------
   7 |     # Choosing a tag where the Android build tools match what we have in android/build.gradle to avoid the
   8 |     # container having to download them
   9 | >>> FROM reactnativecommunity/react-native-android:v13.2.1 AS builder
  10 |     
  11 |     WORKDIR /reactnativesrc/
--------------------
ERROR: failed to solve: reactnativecommunity/react-native-android:v13.2.1: no match for platform in manifest: not found

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/j8czacicufv0xervxcm6cw3kh
make: *** [reactnative-build-android] Error 1
```

Reply:
huh that one's strange, it seems stuck at grabbing the base image which should exist:
https://hub.docker.com/layers/reactnativecommunity/react-native-android/v13.2.1/images/sha256-9b94b331a83dceb53a3e6bb3b4afb4d29bd372ac78e18d33241199e6d2dad5d6?context=explore

I wonder if it's a host architecture difference, taking a look at some of the workarounds from https://stackoverflow.com/questions/76355896/docker-build-platform-linux-amd64-fails-error-failed-to-solve-no-match-for, you could try:
```
docker build -f src/reactnativeapp/android.Dockerfile --platform=linux/amd64 --output=. src/reactnativeapp
```

I wonder if that `platform` flag is needed for compatibility

Reply:
yes, platform flag solved the issue

Reply:
It took 16 minutes to execute on my M1 Max with 64GB of memory. I believe it will take longer in my Ubuntu with i5 7th gen 😅

Reply:
Also, not quite sure how that will help in the current state of the app, as we can't configure the frontend and Collector endpoints.

Reply:
Good point!
Maybe we can keep it, and then just point users to this page: https://developer.android.com/studio/run/emulator-install-add-files

So, for now, they would use the dockerfile to build the APK and Android Studio to run it in an emulated device locally.

Later we change that to something like "install in your personal phone".

Reply:
I'm okay in keeping it as is

Reply:
sounds good, I added that link to the README and that extra flag for the Docker command here: https://github.com/open-telemetry/opentelemetry-demo/pull/1781/commits/42af50d93b6fc6a3d6152f52c1e32919fe788ae1